### ISSUE TITLE
iscsi redirect - missleading error messages "login authentication failure"

### ISSUE THREAD

**Issue by labazza** [open-iscsi/open-iscsi#117]
it has been observed with various iscsi targets, making use of redirect feature and chap authentication, that while establishing a session with the target virtual IP and being redirected to the actual node IP the daemon would log:
```
Login authentication failed with target....
Kernel reported iSCSI connection 14:0 error (1020 - ISCSI_ERR_TCP_CONN_CLOSE: TCP connection closed) state (1)
```
Followed by an authentication successful and connection established.
Albeit can be considered a cosmetic issue and expected behaviour this is reported as an error in the log and is not possible to understand to which node IP is connecting.
Is there any plan in better integrate the iscsi redirect behaviour with authentication?

**Comment by gonzoleeman** [comment]
Since this is cosmetic, the priority is not very high, though that would change if you supplied a suggested patch.

**Comment by gonzoleeman** [comment]
Is this still an issue? I do not have any hardware targets, so I can't current implement redirection testing. But if this is still an issue I'd like to address it.

**Comment by fabiomirmar** [comment]
Hi!

Yes, this is still an issue. You can setup some redirection testing with the following steps: https://pastebin.ubuntu.com/p/Vzfmw7RWwJ/

Thanks for looking into this.

Cheers,
Fabio Martins

**Comment by gonzoleeman** [comment]
I spent a bit of time trying to reproduce this, to no avail.

The iscsitarget package is outdated and unsupported on current kernels.

The normal current Linux target, LIO, does not support redirect.

I managed to get STGT (tgtd) working with redirect on openSUSE Tumbleweed, but it seems to have an issue with CHAP authentication, so I couldn't reproduce your issue.

If you are willing to help me test, perhaps we can still figure out the issue?

If you run iscsid and iscsiadm with "-d 8" and capture the output, I'd like to see what's happening around the error messages you report.

**Comment by fabiomirmar** [comment]
Sure, here is the output of iscsid with -d 8: https://pastebin.ubuntu.com/p/TnXbztn5MC/

And here the iscsiadm with -d 8: https://pastebin.ubuntu.com/p/w9qFVdJ45s/

**Comment by gonzoleeman** [comment]
Yes, in this context, this is not an error message per se. When open-iscsi gets a redirect, it closes the session, as if the login didn't work, but with an error code of "redirect". So it uses the "retry" mechanism by first resetting the target portal, then failing the connection. So this is not a problem.

**Comment by fabiomirmar** [comment]
Thanks. Yes, I’m aware of it. The ask is actually to improve the error message, so the user knows that this is not actually an error, without having to enable debugging or collect a tcpdump to actually understand what’s going on.

**Comment by gonzoleeman** [comment]
No, I do not see a simple way of telling the transport that, in the case of redirect, it's okay for the connection to be closed by the other end, so don't print a message. The problem is that the transport would have to look inside the iSCSI PDU to get the status out of the login response.

I'm open to a fix that would quiet the message(s) in this case, but at the very least it seems like a bit of work to remove the message.
